src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -26,50 +28,84 @@
 @Lazy
 @Component
 public class TreasuryOpiIngestionActivityImpl implements TreasuryOpiIngestionActivity {
-  private final IngestionFlowFileType ingestionflowFileType;
-  private final IngestionFlowFileDao ingestionFlowFileDao;
-  private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final IngestionFlowFileType ingestionflowFileType;
+    private final IngestionFlowFileDao ingestionFlowFileDao;
+    private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final TreasuryUnmarshallerService treasuryUnmarshallerService;
+
+
+    public TreasuryOpiIngestionActivityImpl(
+                                            IngestionFlowFileDao ingestionFlowFileDao,
+                                            IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService,
+                                            TreasuryUnmarshallerService treasuryUnmarshallerService) {
+        this.ingestionflowFileType = IngestionFlowFileType.OPI;
```

Comment:
this could be a constant of this class instead of instantiate it on the Constructor

Reply:
removed association in costructor and aligned code